packages/Webkul/Velocity/webpack.mix.js

```diff
@@ -1,40 +1,79 @@
-const { mix } = require('laravel-mix');
+const path   = require('path');
+const mix    = require('laravel-mix');
+const colors = require('colors');
 
 require('laravel-mix-merge-manifest');
+require('laravel-mix-clean');
 
-let publicPath = '../../../public/themes/velocity/assets';
+const prodPublicPath = path.join('publishable', 'assets');
+const devPublicPath  = path.join('..', '..', '..', 'public', 'themes', 'velocity', 'assets');
+const publicPath     = mix.inProduction() ? prodPublicPath : devPublicPath;
 
-if (mix.inProduction()) {
-    publicPath = 'publishable/assets';
-}
+console.log(colors.bold.blue(`Assets will be published in: ${publicPath}`));
 
-mix.setPublicPath(publicPath).mergeManifest();
-mix.disableNotifications();
+const assetsPath = path.join(__dirname, 'src', 'Resources', 'assets');
+const jsPath     = path.join(assetsPath, 'js');
+const imagesPath = path.join(assetsPath, 'images');
 
 mix
-    .js(__dirname + '/src/Resources/assets/js/app-core.js', 'js/velocity-core.js')
-    .js(
-        __dirname + '/src/Resources/assets/js/app.js',
-        'js/velocity.js'
+    .setPublicPath(publicPath)
+
+    .js(path.join(jsPath, 'app-core.js'), 'js/velocity-core.js')
+    .js(path.join(jsPath, 'app.js'), 'js/velocity.js')
+    .vue()
+
+    .alias({
+               '@Components': path.join(jsPath, 'UI', 'components')
+           })
+    .extract({
+                 to: `/js/components.js`,
+                 test(mod){ return /(component|style|loader|node)/.test(mod.nameForCondition()) }
+             }
```

Comment:
Alright, can you tell me the flow of this portion because then I will check for mobile also?

Reply:
As I am seeing this is adding all the components.

Reply:
Alias is just a shortcut that you can use in every script bundled by webpack (I'm thinking about naming it `@Components\Velocity`, it may avoid confusion in the future.

Reply:
I know about this and currently, I am also busy on some other task. Otherwise, I am will try to use reduce all the components from the main app and use it when it is actually needed.

But that needs time and analysis.

Reply:
It's not very clear to me if the Velocity package is the one which handles every Vue component, or it's just about the Velocity theme, and Vue components can be used in every package. 

Can you tell me more about this? thanks